yarl/_url.py

```diff
@@ -206,89 +264,20 @@ def __new__(
         if strict is not None:  # pragma: no cover
             warnings.warn("strict parameter is ignored")
         if type(val) is str:
-            val = split_url(val)
+            split_result, cache = encode_url(val)
         elif type(val) is cls:
             return val
         elif type(val) is SplitResult:
             if not encoded:
                 raise ValueError("Cannot apply decoding to SplitResult")
+            split_result = val
+            cache = {}
         elif isinstance(val, str):
-            val = split_url(str(val))
+            split_result, cache = encode_url(str(val))
```

Comment:
I think we also need to handle the case where its already encoded,  it looks like we have no coverage for that